### DISCUSSION TITLE
rtxmlrpc load.start or load.normal gives me an error "Could not find info-hash"

### DISCUSSION THREAD

**Discussion by govardha** [kannibalox/pyrosimple#43] [Q&A] [answered]
Hello,

I am trying to get into pyrosimple gradually.  When I run the following command

`rtxmlrpc load.start 'some public meta.tracker'` 

I get the following error.

`ERROR:pyrosimple.scripts.rtxmlrpc.RtorrentXmlRpc:While calling load.normal('blah.torrent') via <RTorrentProxy via xml for scgi+unix:///var/run/blahblah/.rtorrent.sock>: <HashNotFound -404: 'Could not find info-hash.'>`

When I did an lstor on the meta file I get the following.

```
NAME something.mkv
SIZE 3.0 GiB (780 * 4.0 MiB + 2.7 MiB)
META 3.0 GiB (pieces 15.3 KiB 0.0%)
HASH 59A896B44EF1608AC7DC3C366239E275A970E1F2
URL  udp://tracker.openbittorrent.com:80/announce
PRV  NO (DHT/PEX enabled)
TIME 2023-06-29 23:17:21
BY   uTorrent/3.6

FILE LISTING
some file name
```

What am I doing wrong?  When I load the same meta file using flood or rutorrent it loads up just fine into rtorrent.

Thank you!

**Comment by kannibalox** [comment]
I'm a bit confused as to how it's getting a HashNotFound error, but I suspect the crux of the problem is that since rtxmlrpc just sends commands directly to rTorrent, an empty argument needs to be provided at the beginning:
```
rtxmlrpc load.start '' 'some public meta.tracker'
```

**Reply by govardha** [reply]
That worked, but I am still confused about the usage of that command.

I used this link trying to figure things out.

https://rtorrent-docs.readthedocs.io/en/latest/cmd-ref.html#term-load-start-verbose

I would appreciate any clarification on what I did wrong though.

**Reply by kannibalox** [reply]
To quote from the top of that page:
> All d.* commands take an info hash as the first argument when called over the XML-RPC API, to uniquely identify the target object. ‘Target’ is the term used for that 1st parameter in error messages and so on.
> 
>     d.name = ‹hash› ≫ string ‹name›
> 
> When called within configuration methods or in a Ctrl-X prompt, the target is implicit. It is explicit and must be provided for all XMLRPC calls, with very few exceptions like the xmlrpc-c built-ins.

Where your confusion is probably arising from is that rtorrent-docs lists commands in native syntax, where the first argument is pretty much always implied. All commands issued over RPC (whether rtxmlrpc or something else) still need to specify the first argument.

On a side note, can you tell what version of rTorrent you're using? I'd like to try and reproduce that HashNotFound error

**Reply by govardha** [reply]
Hello,

I am on v0.9.8 from here https://github.com/jesec/rtorrent/releases/latest/download/rtorrent-linux-amd64 which I believe you recommended in your docs.

Thank you!

**Reply by kannibalox** [reply]
Cool, and I'd still recommend it. I'm guessing that the `blah.torrent` filename in the error message is just a placeholder, and whatever filename was present just happened to be exactly 40 characters in length, which caused rTorrent to try and read it as an infohash.

None of this has any bearing on the problem you had or the fix, I just like solving mysteries :smile:

**Reply by govardha** [reply]
I just noticed, when I tried the command I get a proper return code of 0 but I don't see the torrent loaded.  Should I have to set the data dir and other stuff?

**Reply by kannibalox** [reply]
Try the `*throw` variants, e.g.
```bash
rtxmlrpc load.throw '' 'some public meta.tracker'
```

**Reply by govardha** [reply]
Ok I tried this and unfortunately doesn't work., although the file most definitely does exist.  

`
boo@bar:~/Dropbox/etoile$ rtxmlrpc  load.throw '' "blah.torrent"
ERROR:pyrosimple.scripts.rtxmlrpc.RtorrentXmlRpc:While calling load.throw('', 'blah.torrent') via <RTorrentProxy via xml for scgi+unix:///var/run/boo/.rtorrent.sock>: <RpcError -503: 'Failed to open file.'>
`

**Reply by kannibalox** [reply, marked as answer]
If the file exists, two common things to check for are if rTorrent has access to read it, and if you're using a relative path instead of an absolute one (or you can check the current working directory of rTorrent with `system.cwd`, but generally it's just easy to feed it absolute paths for the load commands).

**Comment by govardha** [comment]
And kannibalox saves the day again, thank you once again!.

`rtxmlrpc load.start '' $(pwd)/foo.torrent`
